UMA-Bots/Large-Relayer-Token-Balance-Change/src/agent.ts

```diff
@@ -0,0 +1,61 @@
+import { Finding, HandleTransaction, ethers, Initialize, TransactionEvent, getEthersProvider } from "forta-agent";
+import { NetworkManager } from "forta-agent-tools";
+import { TRANSFER_EVENT, getFindingInstance, Dictionary, loadLruCacheData } from "./utils";
+import { NetworkDataInterface, NM_DATA } from "./network";
+import LRU from "lru-cache";
+import { BigNumber } from "ethers";
+
+const networkManager = new NetworkManager(NM_DATA);
+const thisLru = new LRU<string, Dictionary<string>>({ max: 1000 }); // token address => { wallet address => balance }
+
+export function provideInitialize(
+  networkManager: NetworkManager<NetworkDataInterface>,
+  provider: ethers.providers.Provider,
+  passedLru: LRU<string, Dictionary<string>>
+): Initialize {
+  return async () => {
+    await networkManager.init(provider);
+    await loadLruCacheData(networkManager, provider, passedLru);
+  };
+}
+
+export function provideHandleTransaction(
+  transferEvent: string,
+  networkManager: NetworkManager<NetworkDataInterface>,
+  passedLru: LRU<string, Dictionary<string>>
+): HandleTransaction {
+  return async (txEvent: TransactionEvent) => {
+    let alertThreshold = networkManager.get("alertThreshold");
+    const findings: Finding[] = [];
+    const transferEventTxns = txEvent.filterLog(transferEvent, networkManager.get("monitoredTokens")); // Transfer event transactions for monitored addresses
+    transferEventTxns.forEach((transferEvent) => {
+      const { from, to, value } = transferEvent.args;
+      let valueBN: BigNumber = BigNumber.from(value);
+
+      if (networkManager.get("monitoredAddresses").includes(from)) {
+        let prevBalance = BigNumber.from(passedLru.get(transferEvent.address)![from]);
+        if (valueBN.gte(prevBalance.mul(alertThreshold).div(100))) {
```

Comment:
```suggestion
        if (valueBN.mul(100).gte(prevBalance.mul(alertThreshold))) {
```

Avoiding the division (applicable to L46 too)